sdk/core/System.ClientModel/tests/Auth/TOkenProviderTests.cs

```diff
@@ -0,0 +1,289 @@
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using NUnit.Framework;
+
+namespace System.ClientModel.Tests.Auth;
+
+public class TokenProviderTests
+{
+    [Test]
+    public void SampleUsage()
+    {
+        // usage for TokenProvider abstract type
+        TokenProvider<IScopedToken> provider = new ScopedTokenProvider<IScopedToken>();
+        var client = new FooClient(new Uri("http://localhost"),  provider);
+
+        // usage for TokenProvider2 abstract type
+        ITokenProvider provider2 = new ScopedTokenProvider2();
+        client = new FooClient(new Uri("http://localhost"),  provider);
+
+        // usage for policy only and no public tokenProvider abstraction.
+        var policy = new ScopedAuthenticationPolicy(provider, "scope");
+        client = new FooClient(new Uri("http://localhost"), policy);
+    }
+
+    public class FooClient
+    {
+        public FooClient(Uri uri, ApiKeyCredential credential)
+        {
+            ClientPipeline pipeline = ClientPipeline.Create(new(),
+            perCallPolicies: ReadOnlySpan<PipelinePolicy>.Empty,
+            perTryPolicies: [ApiKeyAuthenticationPolicy.CreateBasicAuthorizationPolicy(credential)],
+            beforeTransportPolicies: ReadOnlySpan<PipelinePolicy>.Empty);
+        }
+
+        // Not ideal because the ctor must have a specific token provider type due to generic invariance.
+        public FooClient(Uri uri, TokenProvider<IScopedToken> credential)
+        {
+            ClientPipeline pipeline = ClientPipeline.Create(new(),
+            perCallPolicies: ReadOnlySpan<PipelinePolicy>.Empty,
+            perTryPolicies: [AuthenticationPolicyFactory.CreateAuthenticationPolicy(credential, "myScope")],
+            beforeTransportPolicies: ReadOnlySpan<PipelinePolicy>.Empty);
+        }
+
+        public FooClient(Uri uri, ITokenProvider credential)
+        {
+            ClientPipeline pipeline = ClientPipeline.Create(new(),
+            perCallPolicies: ReadOnlySpan<PipelinePolicy>.Empty,
+            perTryPolicies: [AuthenticationPolicyFactory.CreateAuthenticationPolicy(credential, "myScope")],
+            beforeTransportPolicies: ReadOnlySpan<PipelinePolicy>.Empty);
+        }
+
+        // Generated from the TypeSpec spec.
+        private readonly IReadOnlyDictionary<string, object> context = new Dictionary<string, object> {
+            { "scopes", new string[] { "myScope_from_spec" } },
+            { "authorizationUrl" , "https://myAuthserver.com/token"} };
+
+        public FooClient(Uri uri, OAuthPipelinePolicy authPolicy)
+        {
+            ClientPipeline pipeline = ClientPipeline.Create(new(),
+            perCallPolicies: ReadOnlySpan<PipelinePolicy>.Empty,
+            perTryPolicies: [authPolicy.CreateAuthenticationPolicy(context)],
+            beforeTransportPolicies: ReadOnlySpan<PipelinePolicy>.Empty);
+        }
+    }
+
+    public class ScopedAuthenticationPolicy : OAuthPipelinePolicy
```

Comment:
[Policy Only] example stand-alone policy which encapsulates the TokenProvider implementation and type